src/test/pybind/test_rados.py

```diff
@@ -337,6 +337,72 @@ def test_remove_snap(self):
         self.ioctx.remove_snap('foo')
         eq(list(self.ioctx.list_snaps()), [])
 
+    def test_set_omap(self):
+        keys = ("1", "2", "3")
+        values = ("aaa", "bbb", "ccc")
+        write_op = self.ioctx.create_write_op()
+        self.ioctx.set_omap(write_op, keys, values)
+        self.ioctx.operate_write_op(write_op, "hw")
+        self.ioctx.release_write_op(write_op)
+        read_op = self.ioctx.create_read_op()
+        iter,ret = self.ioctx.get_omap_vals(read_op, "", "", 3)
+        self.ioctx.operate_read_op(read_op,"hw")
+        pair = iter.next()
+        pair = iter.next()
+        eq(pair.key, "2")
+        eq(pair.value, "bbb")
+        self.ioctx.release_read_op(read_op)
+
+    def test_get_omap_vals_by_keys(self):
+        keys = ("1", "2", "3")
+        values = ("aaa", "bbb", "ccc")
+        write_op = self.ioctx.create_write_op()
+        self.ioctx.set_omap(write_op, keys, values)
+        self.ioctx.operate_write_op(write_op, "hw")
+        self.ioctx.release_write_op(write_op)
+        read_op = self.ioctx.create_read_op()
+        iter,ret = self.ioctx.get_omap_vals_by_keys(read_op,("3",))
+        self.ioctx.operate_read_op(read_op,"hw")
+        pair = iter.next()
+        eq(pair.key, "3")
+        eq(pair.value, "ccc")
+        self.ioctx.release_read_op(read_op)
+
+    def test_get_omap_keys(self):
+        keys = ("1", "2", "3")
+        values = ("aaa", "bbb", "ccc")
+        write_op = self.ioctx.create_write_op()
+        self.ioctx.set_omap(write_op, keys, values)
+        self.ioctx.operate_write_op(write_op, "hw")
+        self.ioctx.release_write_op(write_op)
+        read_op = self.ioctx.create_read_op()
+        iter,ret = self.ioctx.get_omap_keys(read_op,"",2)
+        self.ioctx.operate_read_op(read_op,"hw")
+        pair = iter.next()
+        eq(pair.key, "1")
+        eq(pair.value, None)
+        pair = iter.next()
+        eq(pair.key, "2")
+        eq(pair.value, None)
+        self.ioctx.release_read_op(read_op)
+
+    def test_clear_omap(self):
+        keys = ("1", "2", "3")
+        values = ("aaa", "bbb", "ccc")
+        write_op = self.ioctx.create_write_op()
+        self.ioctx.set_omap(write_op, keys, values)
+        self.ioctx.operate_write_op(write_op, "hw")
+        self.ioctx.clear_omap(write_op)
+        self.ioctx.operate_write_op(write_op, "hw")
```

Comment:
it's not generally safe to use the same write_op for more than one operate_write_op() call. The same goes for read ops. It'd be good to have a test for multiple ops in the same write_op here though.